sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -21,149 +21,54 @@
 # SOFTWARE.
 
 
-
 """Http Logging Policy for Azure SDK"""
 
-import os
-import urllib
+import json
 import logging
-import types
-import ast
+import time
+from typing import Optional
 
+from azure.core.pipeline import PipelineRequest, PipelineResponse, HTTPRequestType, HTTPResponseType
 from azure.core.pipeline.policies import HttpLoggingPolicy
+from .http_constants import HttpHeaders
 
-
-
+def _format_error(payload: str) -> str:
+    output = json.loads(payload)
+    return output['message'].replace("\r", " ")
 class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
 
-    def __init__(self, logger=None, **kwargs): # pylint: disable=unused-argument
-        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", None)
+    def __init__(self, logger: Optional[logging.Logger] = None, **kwargs): # pylint: disable=unused-argument
+        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", False)
         super().__init__(logger, **kwargs)
-        if self._enable_diagnostics_logging:
-            self.logger = logger
+        self.logger = logger or logging.getLogger(
+            "azure.cosmos._cosmos_http_logging_policy"
+        )
+        cosmos_disallow_list = ["Authorization", "ProxyAuthorization"]
+        cosmos_allow_list = [
+            v for k, v in HttpHeaders.__dict__.items() if not k.startswith("_") and k not in cosmos_disallow_list
+        ]
+        self.allowed_header_names = set(cosmos_allow_list)
 
     def on_request(self, request): # pylint: disable=too-many-return-statements, too-many-statements
         # type: (PipelineRequest) -> None
-        http_request = request.http_request
         options = request.context.options
         self._enable_diagnostics_logging = request.context.setdefault(
             "enable_diagnostics_logging",
```

Comment:
We don't need this - as discussed earlier, because of the lack of thread-safety here, we can't override this on a per-method basis